datastream-server-restli/src/main/java/com/linkedin/datastream/server/diagnostics/ServerHealthResources.java

```diff
@@ -106,7 +114,16 @@ private TaskHealthArray buildTasksHealthForConnectorType(String connectorType) {
 
       taskHealth.setName(task.getDatastreamTaskName());
       taskHealth.setPartitions(task.getPartitions().toString());
-      taskHealth.setSourceCheckpoint(task.getCheckpoints().toString());
+      if (_checkpointResources.isCustomCheckpointing(connectorType)) {
+        try {
+          taskHealth.setSourceCheckpoint(_checkpointResources.getKafkaCustomCheckpointProvider(task.getDatastreams().get(0)).getSafeCheckpoint().toString());
+        } catch (Exception e) {
+          _errorLogger.logAndThrowRestLiServiceException(HttpStatus.S_400_BAD_REQUEST,
```

Comment:
Is it going be always bad request that would hit this case?
Are there possible other errors that should return other http response code?